pkgs/by-name/ni/niri/package.nix

```diff
@@ -38,56 +47,76 @@ rustPlatform.buildRustPackage rec {
     };
   };
 
+  strictDeps = true;
+
   nativeBuildInputs = [
+    clang
     pkg-config
     rustPlatform.bindgenHook
-    autoPatchelfHook
-    clang
-  ];
-
-  buildInputs = [
-    wayland
-    systemd # For libudev
-    seatd # For libseat
-    libdisplay-info
-    libxkbcommon
-    libinput
-    mesa # For libgbm
-    fontconfig
-    stdenv.cc.cc.lib
-    pipewire
-    pango
   ];
 
-  runtimeDependencies = [
-    wayland
-    mesa
-    libglvnd # For libEGL
-  ];
+  buildInputs =
+    [
+      libdisplay-info
+      libglvnd # For libEGL
+      libinput
+      libxkbcommon
+      mesa # For libgbm
+      pango
+      seatd
+      wayland # For libwayland-client
+    ]
+    ++ lib.optional (withDbus || withScreencastSupport || withSystemd) dbus
+    ++ lib.optional withScreencastSupport pipewire
+    ++ lib.optional withSystemd systemd; # Also includes libudev
```

Comment:
isn't udev required to access input devices? like, is this actually an optional dependency?

Reply:
*Systemd's* libudev is, but yeah we need something when we're not building with that. `eudev` is the alternative AFAIK